### PR TITLE
feat(YouTube - Playback speed): Add speed dialog button to video player

### PR DESCRIPTION
**Pull request by MarcaDian**
Adds a simply speed button to the bottom of video player that called flyout custom speed dialog same as from flyout menu.

<details>

Зображення WhatsApp, дата_ 2024-05-20 о 23 28 57_a81b6303
</details>

Partially closes https://github.com/ReVanced/revanced-patches/issues/1620

integrations  https://github.com/ReVanced/revanced-integrations/pull/636

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Adding too many buttons will create overlaps for example with the chapters.

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/PlaybackSpeedPatch.kt [hunk 1]
```diff
@@ -4,13 +4,19 @@ import app.revanced.patcher.data.BytecodeContext
...
+import app.revanced.patches.youtube.video.speed.button.CustomPlaybackSpeedButtonPatch
 import app.revanced.patches.youtube.video.speed.custom.CustomPlaybackSpeedPatch
 import app.revanced.patches.youtube.video.speed.remember.RememberPlaybackSpeedPatch
 
 @Patch(
     name = "Playback speed",
-    description = "Adds options to customize available playback speeds and to remember the last playback speed selected.",
-    dependencies = [CustomPlaybackSpeedPatch::class, RememberPlaybackSpeedPatch::class],
+    description = "Adds options to customize available playback speeds, to remember the last playback speed selected " +
+            "and to display custom speed dialog button in the video player.",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/PlaybackSpeedPatch.kt:14]:
```suggestion
            "and show a speed dialog button to the video player.",
```

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patches.all.misc.resources.AddResourcesPatch
+import app.revanced.patches.shared.misc.settings.preference.SwitchPreference
+import app.revanced.patches.youtube.misc.playercontrols.PlayerControlsBytecodePatch
+import app.revanced.patches.youtube.misc.settings.SettingsPatch
+import app.revanced.patches.youtube.video.speed.custom.CustomPlaybackSpeedPatch
+
+@Patch(
+    description = "Adds options to display custom speed dialog button in the video player",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt:13]: "Custom" isn't necessary here. The regular speed dialog is shown from YouTube. Please rename the patch as well.

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        PlayerControlsBytecodePatch::class,
+        SettingsPatch::class,
+        AddResourcesPatch::class,
+    ],
+)
+@Suppress("unused")
+object CustomPlaybackSpeedButtonPatch : BytecodePatch(emptySet()) {
+
+    private const val SPEED_BUTTON_CLASS_DESCRIPTOR =
+        "Lapp/revanced/integrations/youtube/videoplayer/CustomSpeedDialogButton;"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt:26]:
```suggestion
    private const val SPEED_BUTTON_CLASS_DESCRIPTOR =
        "Lapp/revanced/integrations/youtube/videoplayer/CustomSpeedDialogButton;"
```

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        CustomPlaybackSpeedPatch::class,
+        PlayerControlsBytecodePatch::class,
+        SettingsPatch::class,
+        AddResourcesPatch::class,
+    ],
+)
+@Suppress("unused")
+object CustomPlaybackSpeedButtonPatch : BytecodePatch(emptySet()) {
+
+    private const val SPEED_BUTTON_CLASS_DESCRIPTOR =
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt:25]: This should be named INTEGRATIONS_CLASS_DESCRIPTOR

**Comment by kazimmt** [comment]
This button will be helpful for non-root users as we encounter this issue: https://github.com/ReVanced/revanced-patches/issues/3203

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt (hunk 2 above)
Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/video/speed/button/CustomPlaybackSpeedButtonPatch.kt]: While it's part of custom speed patch, its technically not related to it. It merely opens the speed dialog.

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/button/PlaybackSpeedButtonPatch.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patches.all.misc.resources.AddResourcesPatch
+import app.revanced.patches.shared.misc.settings.preference.SwitchPreference
+import app.revanced.patches.youtube.misc.playercontrols.PlayerControlsBytecodePatch
+import app.revanced.patches.youtube.misc.settings.SettingsPatch
+import app.revanced.patches.youtube.video.speed.custom.CustomPlaybackSpeedPatch
+
+@Patch(
+    description = "Adds option to display playback speed dialog button in the video player.",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/button/PlaybackSpeedButtonPatch.kt:13]:
```suggestion
    description = "Adds the option to display playback speed dialog button in the video player.",
```

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/PlaybackSpeedPatch.kt [hunk 6]
```diff
@@ -4,13 +4,19 @@ import app.revanced.patcher.data.BytecodeContext
...
+import app.revanced.patches.youtube.video.speed.button.PlaybackSpeedButtonPatch
 import app.revanced.patches.youtube.video.speed.custom.CustomPlaybackSpeedPatch
 import app.revanced.patches.youtube.video.speed.remember.RememberPlaybackSpeedPatch
 
 @Patch(
     name = "Playback speed",
-    description = "Adds options to customize available playback speeds and to remember the last playback speed selected.",
-    dependencies = [CustomPlaybackSpeedPatch::class, RememberPlaybackSpeedPatch::class],
+    description = "Adds options to customize available playback speeds, to remember the last playback speed selected " +
+            "and show a speed dialog button to the video player.",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/PlaybackSpeedPatch.kt:14]:
```suggestion
    description = "Adds options to customize available playback speeds, remember the last playback speed selected " +
            "and show a speed dialog button to the video player.",
```

**Pull request merged**